services/provider/server/server.go

```diff
@@ -189,3 +206,153 @@ func newClient() (client.Client, error) {
 
 	return client, nil
 }
+func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerResource *ocsv1alpha1.StorageConsumer) ([]*pb.ExternalResource, error) {
+	var extR []*pb.ExternalResource
+
+	// Configmap with mon endpoints
+	configmap := &v1.ConfigMap{}
+	err := s.client.Get(ctx, types.NamespacedName{Name: monConfigMap, Namespace: s.namespace}, configmap)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get %s configMap. %v", monConfigMap, err)
+	}
+
+	// Get address of first mon from the monConfigMap configmap
+	cmData := strings.Split(configmap.Data["data"], ",")
+	if len(cmData) == 0 {
+		return nil, fmt.Errorf("configmap %s data is empty", monConfigMap)
+	}
+
+	extR = append(extR, &pb.ExternalResource{
+		Name: monConfigMap,
+		Kind: "ConfigMap",
+		Data: mustMarshal(map[string]string{
+			"data":     cmData[0], // Address of first mon
+			"maxMonId": "0",
+			"mapping":  "{}",
+		})})
+
+	scMon := &v1.Secret{}
+	// Secret storing cluster mon.admin key, fsid and name
+	err = s.client.Get(ctx, types.NamespacedName{Name: monSecret, Namespace: s.namespace}, scMon)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get %s secret. %v", monSecret, err)
+	}
+
+	fsid := string(scMon.Data["fsid"])
+	if fsid == "" {
+		return nil, fmt.Errorf("secret %s data fsid is empty", monSecret)
+	}
+
+	extR = append(extR, &pb.ExternalResource{
+		Name: monSecret,
+		Kind: "Secret",
+		Data: mustMarshal(map[string]string{
+			"fsid":         fsid,
+			"mon-secret":   "mon-secret",
+			"admin-secret": "admin-secret",
+		})})
+
+	// Service for monitoring endpoints
+	scMonitoring := &v1.Service{}
+	err = s.client.Get(ctx, types.NamespacedName{Name: "rook-ceph-mgr", Namespace: s.namespace}, scMonitoring)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get rook-ceph-mgr service. %v", err)
+	}
+
+	if scMonitoring.Spec.ClusterIP == "" || strconv.Itoa(int(scMonitoring.Spec.Ports[0].Port)) == "" {
+		return nil, fmt.Errorf("service rook-ceph-mgr clusterIP or port is empty")
+	}
+
+	extR = append(extR, &pb.ExternalResource{
+		Name: "monitoring-endpoint",
+		Kind: "CephCluster",
+		Data: mustMarshal(map[string]string{
+			"MonitoringEndpoint": scMonitoring.Spec.ClusterIP,
+			"MonitoringPort":     strconv.Itoa(int(scMonitoring.Spec.Ports[0].Port)),
+		})})
+
+	if len(consumerResource.Status.CephResources) == 0 {
+		return nil, fmt.Errorf("ceph resources are empty")
+	}
```

Comment:
This should not be an error, it is ok (from the API server perspective to have an empty `CephResources` list, please remove the if statmenent